src/Elders.Cronus/Hosting/CronusServiceCollectionExtensions.cs

```diff
@@ -14,28 +14,50 @@ namespace Elders.Cronus
     public static class CronusServiceCollectionExtensions
     {
         /// <summary>
-        /// // Adds Cronus core services
+        /// Adds Cronus core services
         /// </summary>
         public static IServiceCollection AddCronus(this IServiceCollection services, IConfiguration configuration)
         {
-            return AddCronus(services, new CronusServicesProvider(services, configuration));
+            return services.AddCronus(configuration, new CronusServicesDiscoveryHandler(services, configuration));
         }
 
         /// <summary>
-        /// // Adds Cronus core services
+        /// Adds Cronus core services
         /// </summary>
-        public static IServiceCollection AddCronus(this IServiceCollection services, CronusServicesProvider cronusServicesProvider)
+        public static IServiceCollection AddCronus(this IServiceCollection services, IConfiguration configuration, ICronusServicesDiscoveryHandler cronusServicesProvider)
```

Comment:
The new parameter here `IConfiguration` must be removed. The configuration is already added by the framework in the IServiceCollection.